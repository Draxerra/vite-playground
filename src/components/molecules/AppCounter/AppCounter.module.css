@import "/src/css/abstracts/media-queries";

.app-counter {
  display: grid;
  place-items: center;
  gap: var(--spacing);
}

.app-counter__text {
  font-size: var(--fs-600);
  @media (--small-viewport) {
    grid-column: span 2;
  }
}

.app-counter__actions {
  display: flex;
  gap: var(--spacing);
}
