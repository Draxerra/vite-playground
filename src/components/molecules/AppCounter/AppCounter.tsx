import React, { FC } from "react";
import classNames from "classnames";

import styles from "./AppCounter.module.css";

import AppButton from "~/components/atoms/AppButton";

type AppCounterProps = {
  decrement: () => void;
  increment: () => void;
} & JSX.IntrinsicElements["div"];

const AppCounter: FC<AppCounterProps> = ({
  decrement,
  increment,
  children,
  className,
  ...props
}) => (
  <div {...props} className={classNames(className, styles["app-counter"])}>
    <span className={styles["app-counter__text"]}>
      Current Count: {children}
    </span>
    <AppButton aria-label="Increment value" onClick={increment}>
      Increment
    </AppButton>
    <AppButton aria-label="Decrement value" onClick={decrement}>
      Decrement
    </AppButton>
  </div>
);
export default AppCounter;
