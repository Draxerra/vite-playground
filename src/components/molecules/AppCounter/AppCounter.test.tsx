import React from "react";
import AppCounter from "./AppCounter";
import { screen, render } from "@testing-library/react";
import userEvent from "@testing-library/user-event";

describe("App Counter", () => {
  it("should match snapshot", () => {
    const decrement = jest.fn();
    const increment = jest.fn();
    const counter = render(
      <AppCounter decrement={decrement} increment={increment}>
        10
      </AppCounter>
    ).asFragment();
    expect(counter).toMatchSnapshot();
  });

  it("should fire decrement on counter-decrement click", () => {
    const decrement = jest.fn();
    const increment = jest.fn();
    render(
      <AppCounter decrement={decrement} increment={increment}>
        10
      </AppCounter>
    );
    const decrementBtn = screen.getByText("Decrement");
    userEvent.click(decrementBtn);
    expect(decrement).toHaveBeenCalledTimes(1);
  });

  it("should fire icrement on counter-icrement click", () => {
    const decrement = jest.fn();
    const increment = jest.fn();
    render(
      <AppCounter decrement={decrement} increment={increment}>
        10
      </AppCounter>
    );
    const incrementBtn = screen.getByText("Increment");
    userEvent.click(incrementBtn);
    expect(increment).toHaveBeenCalledTimes(1);
  });
});
