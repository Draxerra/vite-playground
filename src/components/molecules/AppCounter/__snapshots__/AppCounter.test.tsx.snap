// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Counter should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="app-counter"
  >
    <span
      aria-live="polite"
      class="app-counter__text"
      role="region"
    >
      Current Count: 10
    </span>
    <button
      aria-label="Increment counter"
      class="app-button"
    >
      Increment
    </button>
    <button
      aria-label="Decrement counter"
      class="app-button"
    >
      Decrement
    </button>
  </div>
</DocumentFragment>
`;
